flow360/component/simulation/validation/validation_simulation_params.py

```diff
@@ -264,3 +267,55 @@ def _validate_cht_has_heat_transfer(params):
                     )
 
     return params
+
+
+def _check_complete_boundary_condition_and_unknown_surface(params):
+    ## Step 1: Get all boundaries patches from asset cache
+    asset_boundary_entities = params.private_attribute_asset_cache.boundaries
+    if asset_boundary_entities is None or asset_boundary_entities == []:
+        return params
+    asset_boundaries = {boundary.name for boundary in asset_boundary_entities}
+
+    ## Step 2: Collect all used boundaries from the models
+    if len(params.models) == 1 and isinstance(params.models[0], Fluid):
+        return params
```

Comment:
if there are no BC defined at all, why don't we raise?

Reply:
because even if user composes a SimulationParams for meshing _only_ this validator will still get triggered. The context validator does not control when validator is called.

Reply:
Changed this validator to adapt with current validation context. We only raise if we have CASE ValidationContext